registry/servicediscovery/service_instances_changed_listener_impl.go

```diff
@@ -104,6 +122,10 @@ func (lstn *ServiceInstancesChangedListenerImpl) OnEvent(e observer.Event) error
 			newRevisionToMetadata[revision] = metadataInfo
 		}
 		lstn.revisionToMetadata = newRevisionToMetadata
+		for revision, metadataInfo := range newRevisionToMetadata {
+			metaCache.Set(revision, metadataInfo)
+		}
+		metaCache.DumpCache()
```

Comment:
I see the `DumpCache()` is called every time metadata is updated in your lastest patch. I'm worried about performance if the matadata is update frequently. How much persistent metadata improves performance? I am thinking it not costs too much resources to reload the metadata from remotes once the matadata is lost. Wdyt? /cc @chickenlj